_layouts/guide.html

```diff
@@ -31,15 +38,24 @@
   {% if content contains book_tag %}
     {% assign header_orig = book_tag %}
     {% assign header_mod = book_tag | prepend: header_html %}
+    {% assign toc_mod = toc_tag | append: role_main %}
   {% else %}
     {% assign header_orig = nav_tag %}
-    {% assign header_mod = nav_tag | prepend: header_html %}
+    {% assign header_mod = nav_tag | prepend: header_html | append: role_nav %}
+    {% assign ref_mod = ref_tag | append: role_main %}
+    {% assign chap_mod = chap_tag | append: role_main %}
+    {% assign part_mod = part_tag | append: role_main %}
+    {% assign toc_mod = toc_tag %}
   {% endif %}
 
 {%
   endcapture
 %}{{ content
   | replace_first: html_tag, html_mod
   | replace_first: head_tag, head_mod
+  | replace_first: ref_tag, ref_mod
+  | replace_first: chap_tag, chap_mod
+  | replace_first: part_tag, part_mod
+  | replace_first: toc_tag, toc_mod
```

Comment:
All the above changes in this guide rewriting file are a bit hackish.

Ideally, it would probably check for the existence of a tag before replacing it. It would be more involved, however, and I checked to ensure this works.

(In this implementation, it replaces the tags with mods, which may be modified versions or not. If values are blank, it just replaces blank with blank. I verified it works for all variants of the guide pages.)

Reply:
Note: Liquid (and nearly everything you have to do in Liquid) is hackish due to constraints... but it's Jekyll's templating language so it is what it is.